Eggstravaganza/Assets/Scripts/ServerStartup.cs

```diff
@@ -0,0 +1,251 @@
+    {
+        Debug.Log($"OnAllocation: {allocation.AllocationId}");
+        if (string.IsNullOrEmpty(allocation.AllocationId)) return;
+        m_AllocationId = allocation.AllocationId;
+    }
+
+    private async Task<string> AwaitAllocationId()
+    {
+        var config = m_MultiplayService.ServerConfig;
+        Debug.Log("Awaiting allocation. Server config is:\n" +
+                  $"-ServerID: {config.ServerId}\n" +
+                  $"-AllocationID: {config.AllocationId}\n" +
+                  $"-Port: {config.Port}\n" +
+                  $"-QPort: {config.QueryPort}" +
+                  $"-logs: {config.ServerLogDirectory}");
+
+        while (string.IsNullOrEmpty(m_AllocationId))
+        {
+            var configId = config.AllocationId;
+            if (!string.IsNullOrEmpty(configId) && string.IsNullOrEmpty(m_AllocationId))
+            {
+                m_AllocationId = configId;
+                break;
+            }
+
+            await Task.Delay(100);
+        }
+
+        return m_AllocationId;
+    }
+    
+    private async Task<MatchmakingResults> GetMatchmakerAllocationPayloadAsync()
+    {
+        try
+        {
+            var payloadAllocation =
+                await MultiplayService.Instance.GetPayloadAllocationFromJsonAs<MatchmakingResults>();
+            var modelAsJson = JsonConvert.SerializeObject(payloadAllocation, Formatting.Indented);
+            Debug.Log($"Matchmaker payload allocation: {modelAsJson}");
+            return payloadAllocation;
+        }
+        catch (Exception ex)
+        {
+            Debug.LogWarning($"Something went wrong trying to get the Matchmaker payload:\n{ex}");
+        }
+
+        return null;
+    }
+    
+    private async Task StartBackfill(MatchmakingResults payload)
+    {
+        var backfillProperties = new BackfillTicketProperties(payload.MatchProperties);
+        m_LocalBackfillTicket = new BackfillTicket
+            { Id = payload.MatchProperties.BackfillTicketId, Properties = backfillProperties };
+        await BeginBackfilling(payload);
+    }
+
+    private async Task BeginBackfilling(MatchmakingResults payload)
+    {
+        var matchProperties = payload.MatchProperties;
+
+        Debug.Log("Begin backfilling!");
+
+        if (string.IsNullOrEmpty(m_LocalBackfillTicket.Id))
+        {
+            m_CreateBackfillTicketOptions = new CreateBackfillTicketOptions
+            {
+                Connection = m_ExternalConnectionString,
+                QueueName = payload.QueueName,
+                Properties = new BackfillTicketProperties(matchProperties)
+            };
+
+            m_LocalBackfillTicket.Id = await MatchmakerService.Instance.CreateBackfillTicketAsync(m_CreateBackfillTicketOptions);
+        }
+
+        m_Backfilling = true;
+#pragma warning disable 4014
+        BackfillLoop();
+#pragma warning restore 4014
+    }
+
+    private async Task BackfillLoop()
+    {
+        while (m_Backfilling && NeedsPlayers())
```

Comment:
prob for our game we can just do a little `&& !inGame` kind of thing if that might be easier than changing the existing game logic?